addons/barfiller/barfiller.lua

```diff
@@ -0,0 +1,118 @@
+]]
+
+_addon.name = 'BarFiller'
+_addon.author = 'Morath'
+_addon.version = '0.1.5'
+_addon.commands = {'bf','barfiller'}
+_addon.language = 'english'
+
+--Windower Libs
+config = require('config')
+require('functions')
+require('maths')
+require('pack')
+
+--BarFiller Libs
+require('statics')
+
+settings = config.load('data\\settings.xml',default_settings)
+config.register(settings,initialize,bgImage,bgPosX,bgPosY,bgBarHeight,bgBarWidth,fgImage,fgPosX,fgPosY,fgBarHeight,fgBarWidth)
+
+-- Background Bar
+bgImage = settings.bar_settings.background.image
+bgPosX = settings.bar_settings.background.pos.x
+bgPosY = settings.bar_settings.background.pos.y
+bgBarHeight = settings.bar_settings.background.size.height
+bgBarWidth = settings.bar_settings.background.size.width
+
+-- Foreground Bar
+fgImage = settings.bar_settings.foreground.image
+fgPosX = settings.bar_settings.foreground.pos.x
+fgPosY = settings.bar_settings.foreground.pos.y
+fgBarHeight = settings.bar_settings.foreground.size.height
+fgBarWidth = settings.bar_settings.foreground.size.width
+
+-- Background Bar Style
+windower.prim.create('backgroundBar')
+windower.prim.set_position('backgroundBar', getBackgroundPosX(), getBackgroundPosY())
+windower.prim.set_size('backgroundBar', getBackgroundWidth(), getBackgroundHeight())
+windower.prim.set_texture('backgroundBar', bgImage)
+windower.prim.set_fit_to_texture('backgroundBar', false)
+windower.prim.set_repeat('backgroundBar', 1, 1)
+windower.prim.set_visibility('backgroundBar', true)
+
+-- Foreground Bar Style
+windower.prim.create('foregroundBar')
+windower.prim.set_position('foregroundBar', getForegroundPosX(), getForegroundPosY())
+windower.prim.set_size('foregroundBar', getForegroundWidth(), getForegroundHeight())
+windower.prim.set_texture('foregroundBar', fgImage)
+windower.prim.set_fit_to_texture('foregroundBar', false)
+windower.prim.set_repeat('foregroundBar', 1, 1)
+windower.prim.set_visibility('foregroundBar', true)
+
+-- windower.register_event('login',function(name)
+--     windower.send_command('lua i barfiller clear;')
+-- end)
+
+-- Addon commands
+-- Thanks to Byrth & SnickySnacks' BattleMod addon
+windower.register_event('addon command',function(...)
+	local commands = {...}
+	local first_cmd = table.remove(commands,1):lower()
+	if approved_commands[first_cmd] and #commands >= approved_commands[first_cmd].n then
+		if first_cmd == 'clear' or first_cmd == 'c' then		-- Reset EXP bar to 0
+			initialize()
+			print('BarFiller successfully reset the experience counter.')
+		elseif first_cmd == 'reload' or first_cmd == 'r' then	-- Reloads BarFiller
+			windower.send_command('lua r barfiller;')
+		elseif first_cmd == 'unload' or first_cmd == 'u' then	-- Unloads BarFiller
+			windower.send_command('lua u barfiller;')
+		elseif first_cmd == 'help' or first_cmd == 'h' then		-- Display helpful information
+			help()
+		end
+	else
+		print('Missing or invalid command, use "bf help" or "bf h" for list of available commands.')
```

Comment:
May as well call the help function here instead of telling people to call the help function :)